src/Bicep.Core/Syntax/SyntaxHierarchy.cs

```diff
@@ -48,7 +48,7 @@ public bool IsDescendant(SyntaxBase node, SyntaxBase potentialAncestor)
             return false;
         }
 
-        private sealed class ParentTrackingVisitor : SyntaxVisitor
+        private sealed class ParentTrackingVisitor : AstVisitor
```

Comment:
>AstVisitor [](http://example.com/codeflow?start=53&length=10)

The original intent of `SyntaxHierarchy` was to create parent pointers for all the nodes in the CST. Given that all the tests are passing with the change, we are clearly only calculating parents for AST nodes. Worth adding a comment somewhere to indicate that we only create the hierarchy for AST nodes?